vol_link_test.c

```diff
@@ -25962,7 +25962,7 @@ link_iter_external_links_cb(hid_t group_id, const char *name, const H5L_info2_t
         SKIPPED();
         HDprintf("    API functions for external links aren't supported with this "
                  "connector\n");
-        return 1;
```

Comment:
I specifically requested to have all of these callbacks return 1 rather than 0 in order to short-circuit return when a test is skipped. If this is causing these tests to be reported as failures, the issue should be fixed elsewhere.

Reply:
This issue was initially covered up by path issues that I fixed in #54. 

I moved the `SKIPPED()` execution if the flags don't match from the callback to the tests that use the callback, and implemented it in the same branch as #54.